torch/utils/_sympy/interp.py

```diff
@@ -169,6 +170,9 @@ def sympy_interp(
         dtype = torch.bool
     elif isinstance(expr, sympy.Integer):
         dtype = torch.int64
+    elif isinstance(expr, SingletonInt):
```

Comment:
why do we need this?

Per @soulitzer's [comment](https://github.com/pytorch/pytorch/pull/125941#issuecomment-2283112503), I think we can just

(1) when deciding which subclass SymInts to add as extra graph outputs check if the subclass output's size is a nested_int (using `symbolic_shapes.is_nested_int`)

(2) if our output is an nested int, ignore it and don't add it as a graph output

(3) pass in... `None` for its corresponding `outer_size` argument during the `__tensor_unflatten__` call? (@soulitzer was saying that NJT will just ignore the `outer_size` argument, and rely on the `offsets` tensor graph output to construct the nested size).

Reply:
Although, does the 3-liner above also fix the problem / can you explain why? This change doesn't seem intuitive to me, since it looks like the `expr` argument to this function is expected to be a `sympy.Expr`, and not a `SingletonInt`.

Reply:
> why do we need this?

Some of the nested ints were being propagated to this part of the code. By checking if the symint is a nested int, this changed is not needed anymore.